pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -258,49 +257,57 @@ func (exp *flowExporter) sendDataRecord(dataRec ipfix.IPFIXRecord, record flowex
 		case "protocolIdentifier":
 			_, err = dataRec.AddInfoElement(ie, record.Conn.TupleOrig.Protocol)
 		case "packetTotalCount":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.OriginalPackets)
+			if ie.EnterpriseId == ipfixregistry.IANAEnterpriseID {
+				_, err = dataRec.AddInfoElement(ie, record.Conn.OriginalPackets)
+			} else if ie.EnterpriseId == ipfixregistry.IANAReversedEnterpriseID {
+				_, err = dataRec.AddInfoElement(ie, record.Conn.ReversePackets)
+			}
 		case "octetTotalCount":
-			_, err = dataRec.AddInfoElement(ie, record.Conn.OriginalBytes)
-		case "packetDeltaCount":
-			deltaPkts := 0
-			if record.PrevPackets != 0 {
-				deltaPkts = int(record.Conn.OriginalPackets) - int(record.PrevPackets)
+			if ie.EnterpriseId == ipfixregistry.IANAEnterpriseID {
+				_, err = dataRec.AddInfoElement(ie, record.Conn.OriginalBytes)
+			} else if ie.EnterpriseId == ipfixregistry.IANAReversedEnterpriseID {
+				_, err = dataRec.AddInfoElement(ie, record.Conn.ReverseBytes)
 			}
-			if deltaPkts < 0 {
-				klog.Warningf("Delta packets is not expected to be negative: %d", deltaPkts)
+		case "packetDeltaCount":
+			if ie.EnterpriseId == ipfixregistry.IANAEnterpriseID {
+				deltaPkts := 0
+				if record.PrevPackets != 0 {
+					deltaPkts = int(record.Conn.OriginalPackets) - int(record.PrevPackets)
+				}
+				if deltaPkts < 0 {
```

Comment:
are you planning to address it in a future PR?

Reply:
Planned to address it in this PR since it is not a big change.

Reply:
Found that the conversion is necessary if we want to return 0 when `record.PrevPackets > record.Conn.OriginalPackets` and the value when `record.PrevPackets < record.Conn.OriginalPackets`. Because `record.PrevPackets - record.Conn.OriginalPackets` will cause overflow when we want to include the negative value in the warning message.

Reply:
Sorry I'm confused. Which part of the code ensures that we return 0 when `record.PrevPackets > record.Conn.OriginalPackets`?

Reply:
My mistake. we cannot ensure that. The reason to keep the conversion is for warning message to show the negative value. if we don't want to show that info, the code can be simplified as you suggested.

Reply:
We can keep it as is, but I think you should replace the `int()` conversions with `int64()` conversions at least, in case we start offering support for ARM 32-bit architectures

Reply:
Makes sense. Updated. Thanks!